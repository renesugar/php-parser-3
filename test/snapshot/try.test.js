const parser = require("../main");

describe("boolean", () => {
  it("simple", () => {
    expect(
      parser.parseEval(
        "try { call(); } catch (Exception $e) { do_something(); }"
      )
    ).toMatchSnapshot();
  });
  it("finally", () => {
    expect(
      parser.parseEval(
        "try { call(); } catch (Exception $e) { do_something(); } finally { do_something_other(); }"
      )
    ).toMatchSnapshot();
  });
  it("miltiple catch", () => {
    expect(
      parser.parseEval(
        "try { call(); } catch (MyException | MyOtherException $e) { do_something(); }"
      )
    ).toMatchSnapshot();
  });
});
