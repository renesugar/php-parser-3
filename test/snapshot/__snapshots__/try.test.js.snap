// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`boolean finally 1`] = `
Program {
  "children": Array [
    Try {
      "always": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Call {
              "arguments": Array [],
              "kind": "call",
              "what": ClassReference {
                "kind": "classreference",
                "name": "do_something_other",
                "resolution": "uqn",
              },
            },
            "kind": "expressionstatement",
          },
        ],
        "kind": "block",
      },
      "body": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Call {
              "arguments": Array [],
              "kind": "call",
              "what": ClassReference {
                "kind": "classreference",
                "name": "call",
                "resolution": "uqn",
              },
            },
            "kind": "expressionstatement",
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [],
                  "kind": "call",
                  "what": ClassReference {
                    "kind": "classreference",
                    "name": "do_something",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            ClassReference {
              "kind": "classreference",
              "name": "Exception",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`boolean miltiple catch 1`] = `
Program {
  "children": Array [
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Call {
              "arguments": Array [],
              "kind": "call",
              "what": ClassReference {
                "kind": "classreference",
                "name": "call",
                "resolution": "uqn",
              },
            },
            "kind": "expressionstatement",
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [],
                  "kind": "call",
                  "what": ClassReference {
                    "kind": "classreference",
                    "name": "do_something",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            ClassReference {
              "kind": "classreference",
              "name": "MyException",
              "resolution": "uqn",
            },
            ClassReference {
              "kind": "classreference",
              "name": "MyOtherException",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;

exports[`boolean simple 1`] = `
Program {
  "children": Array [
    Try {
      "always": null,
      "body": Block {
        "children": Array [
          ExpressionStatement {
            "expression": Call {
              "arguments": Array [],
              "kind": "call",
              "what": ClassReference {
                "kind": "classreference",
                "name": "call",
                "resolution": "uqn",
              },
            },
            "kind": "expressionstatement",
          },
        ],
        "kind": "block",
      },
      "catches": Array [
        Catch {
          "body": Block {
            "children": Array [
              ExpressionStatement {
                "expression": Call {
                  "arguments": Array [],
                  "kind": "call",
                  "what": ClassReference {
                    "kind": "classreference",
                    "name": "do_something",
                    "resolution": "uqn",
                  },
                },
                "kind": "expressionstatement",
              },
            ],
            "kind": "block",
          },
          "kind": "catch",
          "variable": Variable {
            "byref": false,
            "curly": false,
            "kind": "variable",
            "name": "e",
          },
          "what": Array [
            ClassReference {
              "kind": "classreference",
              "name": "Exception",
              "resolution": "uqn",
            },
          ],
        },
      ],
      "kind": "try",
    },
  ],
  "errors": Array [],
  "kind": "program",
}
`;
